### ISSUE TITLE
get() should not return first Resolver-ed item if multiple items exist, though consider an @Primary to facilitate the same behavior

### ISSUE THREAD

**Issue by xenoterracide** [aurelia/aurelia#871]
currently, this does not throw, I think that it should
```ts
  it('multiple registrations, using get', function () {
    const container = DI.createContainer();
    class Foo {
    }
    container.register(Registration.singleton(Foo, Foo));
    container.register(Registration.singleton(Foo, Foo));
    assert.throws(() => container.get(Foo), /.*Foo.*/, 'contains key');
    assert.throws(() => container.get(Foo), /.*getAll.*/, 'contains getAll');
  });
```

however, it's possible that there should be the ability to register a primary Registration/Resolver, I'm at a loss for exactly how to do this, and what the API should look like, here's an initial thought, and I'm not certain primary makes sense unless you have an interface.

```ts
  it('multiple registrations, using get', function () {
    interface Bar {}
    class Foo implements Bar {
    }
    class Baz implements Bar {
    }
    const Bar = DI.createInterface<Bar>().noDefault();
    const container = DI.createContainer();
 
    container.register(Registration.singleton(Bar, Foo, { primary: true }));
    container.register(Registration.singleton(Bar, Baz));

    container.get(Bar); // Foo
  });
```
additional ways to define might include

```ts
@primary
@singleton
interface Bar {}
class Foo implements Bar {
 }
```

```ts
    interface Bar {}
    class Foo implements Bar {
    }
    class Baz implements Bar {
    }
    const Bar = DI.createInterface<Bar>().withDefault( x => x.singelton(Foo, { singleton: true });
```

**Comment by bigopon** [comment]
I ran into this behavior a few times. It neither throws nor returns what I wanted, and during debugging, it turned out the key used was already registered. Though, it's very rare occasion that I ran into it (mostly during tests), so I think we can kind of leave this to be in "advanced" section (reads poorly doc'd) as it's definitely not common. or do `@primary`/`{ primary: true }` in a way that doesn't unnecessarily harm perf of the 99% case scenario.
